src/Storages/System/StorageSystemDDLWorkerQueue.cpp

```diff
@@ -0,0 +1,75 @@
+#include <algorithm>
+#include <string>
+
+#include "StorageSystemDDLWorkerQueue.h"
+
+#include <Columns/ColumnArray.h>
+#include <DataTypes/DataTypeArray.h>
+#include <DataTypes/DataTypeString.h>
+#include <Interpreters/Context.h>
+#include <Common/ZooKeeper/ZooKeeper.h>
+
+
+namespace DB
+{
+NamesAndTypesList StorageSystemDDLWorkerQueue::getNamesAndTypes()
+{
+    return {
+        {"name", std::make_shared<DataTypeString>()}, // query_<id>
+        {"active", std::make_shared<DataTypeArray>(std::make_shared<DataTypeString>())}, // list of host_fqdn
+        {"finished", std::make_shared<DataTypeArray>(std::make_shared<DataTypeString>())}, // list of host_fqdn
+    };
+}
+
+
+void StorageSystemDDLWorkerQueue::fillData(MutableColumns & res_columns, const Context & context, const SelectQueryInfo &) const
+{
+    zkutil::ZooKeeperPtr zookeeper = context.getZooKeeper();
+
+    String ddl_zookeeper_path = config.getString("distributed_ddl.path", "/clickhouse/task_queue/ddl/");
+    String ddl_query_path;
+
+    // this is equivalent to query zookeeper at the `ddl_zookeeper_path`
+    /* [zk: localhost:2181(CONNECTED) 51] ls /clickhouse/task_queue/ddl
+       [query-0000000000, query-0000000001, query-0000000002, query-0000000003, query-0000000004]
+    */
+    zkutil::Strings queries = zookeeper->getChildren(ddl_zookeeper_path);
+
+    // iterate through queries
+    /*  Dir contents of every query will be similar to
+     [zk: localhost:2181(CONNECTED) 53] ls /clickhouse/task_queue/ddl/query-0000000004
+    [active, finished]
+    */
+    for (const String & q : queries)
+    {
+        /* Fetch all nodes under active & finished.
+        [zk: localhost:2181(CONNECTED) 54] ls /clickhouse/task_queue/ddl/query-0000000004/active
+        []
+       [zk: localhost:2181(CONNECTED) 55] ls /clickhouse/task_queue/ddl/query-0000000004/finished
+       [clickhouse01:9000, clickhouse02:9000, clickhouse03:9000, clickhouse04:9000]
+       */
+        size_t col_num = 0;
+        ddl_query_path = ddl_zookeeper_path + "/" + q;
+        zkutil::Strings active_nodes = zookeeper->getChildren(ddl_query_path + "/active");
```

Comment:
`select ... from system.ddl_worker_queue` may fail with `ZNONODE` in the following cases:
 - Initiator of `ON CLUSTER` query has lost connection [here](https://github.com/ClickHouse/ClickHouse/blob/2e9f10ab3ec8bc99b6b9596e7cb0b9660195fefb/src/Interpreters/DDLWorker.cpp#L1096) and all hosts are busy with older queue entries, so status dirs are not [created](https://github.com/ClickHouse/ClickHouse/blob/2e9f10ab3ec8bc99b6b9596e7cb0b9660195fefb/src/Interpreters/DDLWorker.cpp#L760) yet
 - race with `DDLWorker::runCleanupThread()`

The simplest way to fix it - use `tryGetChildren(...)` and leave some columns empty if we cannot get data from zk. 
Another way is to use [locks in zk](https://github.com/ClickHouse/ClickHouse/blob/2e9f10ab3ec8bc99b6b9596e7cb0b9660195fefb/src/Interpreters/DDLWorker.cpp#L1025), but let's try to do it without locks

Reply:
>The simplest way to fix it - use tryGetChildren(...) and leave some columns empty if we cannot get data from zk.

But this way the data will be unreliable, no?
At least all columns should be marked as `Nullable`, and `Null` means data retrieving error, @tavplubix what do you think?

Reply:
> data will be unreliable

Not precisely, `status` will be `unknown` and and some columns (`query_duration_ms`, `exception`) will be empty. However, i'm not sure if it's possible to distinguish the first case (no one has started to execute entry) with the second (entry are currently removed). 

And of course, we should not provide incomplete data on zk errors other than `ZNONODE`.

Reply:
I have changed to use `tryGetChildren(...)` as suggested.